### PR TITLE
fix race w/ *Conn.requestTimeout

### PR DESCRIPTION
**Pull request by vetinari**
see https://travis-ci.org/go-ldap/ldap/jobs/227891683
cherry picked from #110

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -193,7 +194,9 @@ func (l *Conn) Close() {
 // SetTimeout sets the time after a request is sent that a MessageTimeout triggers
 func (l *Conn) SetTimeout(timeout time.Duration) {
 	if timeout > 0 {
+		l.timeoutMutex.Lock()
 		l.requestTimeout = timeout
+		l.timeoutMutex.Unlock()
```
Comment (liggitt) [inline conn.go]: Might be simpler to make this an atomic value and do lockless load/store. We already have a shim for < go1.3 using locks

**Comment by vetinari** [comment]
using the atomicValue struct triggered the race during travis build...

**Comment by liggitt** [comment]
> using the atomicValue struct triggered the race during travis build...

that's concerning... looks like the 1.3 version needs to hold a pointer to the lock

**Comment by vetinari** [comment]
1.3 was ok, it hit 1.5: https://travis-ci.org/go-ldap/ldap/builds/229842330, the 1.3 failure was a network fetch error while setting up: https://travis-ci.org/go-ldap/ldap/builds/229846728

**Comment by liggitt** [comment]
hmm, that's still concerning, but the atomic load/store int64 is nice and clean and sidesteps that issue. go ahead and squash and remove the unused mutex, then LGTM

**Comment by vetinari** [comment]
hmm... https://travis-ci.org/go-ldap/ldap/jobs/230628746 ... what does the race detector find? concurrent r/w access to the *Conn struct's members?

**Comment by johnweldon** [comment]
@vetinari I think this is handled by the recent merge #134

**Comment by vetinari** [comment]
fixed by #134

**Pull request closed without merging**